src/deepsparse/loggers/function_logger.py

```diff
@@ -31,49 +33,70 @@ class FunctionLogger(BaseLogger):
     according to FunctionLogger's attributes
 
     :param logger: A child DeepSparse Logger object
-    :param target_identifier: The string that needs to match the
-        `identifier` (argument to the log() method),
-        so that the FunctionLogger applies
-        the metric function and logs the data
-    :param function_name: Name of the metric function
-    :param function: The metric function to be applied
-    :param frequency: The frequency with which the metric
-        name is to be applied
+    :param target_logging_configs: A list of TargetLoggingConfig objects
     """
 
     def __init__(
         self,
         logger: BaseLogger,
-        target_identifier: str,
-        function: Callable[[Any], Any],
-        function_name: str = None,
-        frequency: int = 1,
+        target_logging_configs: List["TargetLoggingConfig"],  # noqa F821
     ):
 
         self.logger = logger
-        self.target_identifier = target_identifier
-        self.function_name = function_name or function.__name__
-        self.frequency = frequency
-        self.function = function
-
-        self._function_call_counter = 0
+        self.target_logging_configs = target_logging_configs
+        self._function_call_counter = defaultdict(lambda: defaultdict(int))
 
     def log(self, identifier: str, value: Any, category: MetricCategories):
         """
+        Collect information from the pipeline and:
+        1) Filter it according to the `self.target_logging_configs`
+        2) Apply metric functions according to the `self.target_logging_configs`
+
         :param identifier: The name of the item that is being logged.
         :param value: The data structure that the logger is logging
         :param category: The metric category that the log belongs to
         """
-        extracted_value = match_and_extract(
```

Comment:
let's move this to a helper function, `_match_and_log` or something

Reply:
Please elaborate.